tests/core/arch-x86/stalker-x86.c

```diff
@@ -375,6 +401,239 @@ invoke_flat (TestStalkerFixture * fixture,
+
+  compiled_size_run2 = g_hash_table_size (compiled_run2);
+  g_print ("\tcompiled2: %d\n", compiled_size_run2);
+
+  executed_size_run2 = g_hash_table_size (executed_run2);
+  g_print ("\texecuted2: %d\n", executed_size_run2);
+
+  g_assert_cmpuint (compiled_size_run2, ==, 0);
+  g_assert_cmpuint (executed_size_run2, ==, executed_size_run1);
+
+
+  /* Free resources */
+  g_hash_table_destroy (compiled_run2);
+  g_hash_table_destroy (executed_run2);
+  g_hash_table_destroy (compiled_run1);
+  g_hash_table_destroy (executed_run1);
+
+  gum_stalker_unfollow_me (fixture->stalker);
+
+  g_print ("DONE\n");
+
+}
+
+static void
+prefetch_transformer (GumStalkerIterator * iterator,
+                   GumStalkerOutput * output,
+                   gpointer user_data)
+{
+  const cs_insn * instr;
+  gboolean begin = TRUE;
+  gpointer block_address;
+  gpointer * callout_ctx;
+
+  while (gum_stalker_iterator_next (iterator, &instr))
+  {
+    if (begin)
+    {
+      block_address = GSIZE_TO_POINTER (instr->address);
+      callout_ctx = gum_malloc (sizeof (gpointer));
+      *callout_ctx = block_address;
+
+      gum_stalker_iterator_put_callout (iterator, prefetch_callout,
+          callout_ctx, gum_free);
+
+      if (prefetch_compiled != NULL)
+      {
+        g_hash_table_insert (prefetch_compiled, block_address, block_address);
+      }
+      begin = FALSE;
+    }
+
+    gum_stalker_iterator_keep (iterator);
+  }
+}
+
+static void
+prefetch_callout (GumCpuContext * cpu_context,
+                  gpointer user_data)
+{
+  gpointer * callout_ctx = user_data;
+  gpointer block_address = *callout_ctx;
+
+  if (prefetch_executed != NULL)
+  {
+    g_hash_table_insert (prefetch_executed, block_address, block_address);
+  }
+}
+
+static void
+prefetch_run_child (GumStalker * stalker,
+                    GumMemoryRange * runner_range,
+                    int compile_fd,
+                    int execute_fd)
+{
+  pid_t pid;
+  int ret;
+  int status;
+
+  pid = fork ();
+  g_assert_cmpint (pid, >=, 0);
+  if (pid == 0)
+  {
+    /* child */
+    prefetch_compiled = g_hash_table_new (NULL, NULL);
+    prefetch_executed = g_hash_table_new (NULL, NULL);
+
+    gum_stalker_activate (stalker, prefetch_activation_target);
+    prefetch_activation_target ();
+    pretend_workload (runner_range);
+    gum_stalker_unfollow_me (stalker);
+
+    prefetch_write_blocks (compile_fd, prefetch_compiled);
+    prefetch_write_blocks (execute_fd, prefetch_executed);
+
+    g_hash_table_destroy (prefetch_compiled);
+    prefetch_compiled = NULL;
+    g_hash_table_destroy (prefetch_executed);
+    prefetch_executed = NULL;
+
+    exit (0);
+    return;
+  }
+
+  /* Wait for the child */
+  ret = waitpid (pid, &status, 0);
+  g_assert_cmpint (ret, ==, pid);
+  g_assert_cmpint (WIFEXITED (status), !=, 0);
+  g_assert_cmpint (WEXITSTATUS (status), ==, 0);
+}
+
+__attribute__((always_inline))
+static inline void
```

Comment:
Should be updated to match the example after updating it.

Reply:
Will do.